strawberryfields/utils/post_processing.py

```diff
@@ -0,0 +1,341 @@
+# Copyright 2019-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module provides functions for post-processing samples.
+"""
+import numpy as np
+
+
+def number_expectation_pnr(photon_number_samples, modes=None):
+    """The expectation value of the number operator from PNR samples.
```

Comment:
It might be good to provide the exact observable that is computed. Perhaps something along the lines of

```rest
The expectation value of the number operator from PNR samples.

If applied to a single mode, this simply corresponds to mean photon number
:math:`\langle n_i\rangle`.

For multiple modes, the expectation value of the tensor of number operators,
:math:`\langle n_{i_0} \otimes n_{i_1} \otimes \cdots \otimes n_{i_m}` is returned.
```

Reply:
thanks added it!